### PR TITLE
Benchmark sanitization and minor additions

### PR DESCRIPTION
**Pull request by Izaakwltn**
Addressing issue #1204, this sanitizes the directory while we figure out how to replace `trivial-benchmarks`.
- Changes the benchmarks library to package-per-file
- Removes `trivial-benchmarks` as a dependency

This also adds two benchmarks from [the Benchmarks Game](https://benchmarksgame-team.pages.debian.net/benchmarksgame/index.html), with hopeful additions coming soon.

I think next steps (for future PR(s)) should include a simple Coalton benchmarking framework, ideally without external dependencies, and incorporation of benchmarks into CI.

### CODE REVIEW

File: benchmarks/benchmarks-game/mandelbrot.lisp [hunk 1]
```diff
@@ -0,0 +1,179 @@
+(defpackage #:coalton/benchmarks/benchmarks-game/mandelbrot
```
Comment (stylewarning) [inline benchmarks/benchmarks-game/mandelbrot.lisp:1]: we should write a comment on what this code is benchmarking. i don't mean documenting what mandelbrot is, but rather "this code benchmarks highly polymorphic complex numerics, and high order functions. the code is written in a relatively idiomatic functional style." or so

File: benchmarks/benchmarks-game/mandelbrot.lisp [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+                     -> Integer))
+  (define (escapes? limit fc div? z0)
+    "Returns the number of iterations of the function F "
+    (let count = (cell:new 0))
+    (let result = (cell:new z0))
+
+    (while :escape (< (cell:read count) limit)
+      (cell:update! fc result)
+      (if (div? (cell:read result))
+          (break :escape)
```
Comment (stylewarning) [inline benchmarks/benchmarks-game/mandelbrot.lisp:56]: return should soon work when an open pr is merged within 24h

File: benchmarks/benchmarks-game/mandelbrot.lisp [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+                   -> (Vector Integer)))
+  (define (sample negbound posbound steps limit f div? z0)
+    (let v = (vec:new))
+    (let steps-num = (fromint steps))
+    (let width = (- (real-part posbound) (real-part negbound)))
+    (let height = (- (imag-part posbound) (imag-part negbound)))
+    (let real-offset = (real-part negbound))
+    (let imag-offset = (imag-part negbound))
+    (for bi in (iter:up-to steps-num)
+      (lisp Unit (bi)
```
Comment (stylewarning) [inline benchmarks/benchmarks-game/mandelbrot.lisp:79]: delete

File: benchmarks/benchmarks-game/mandelbrot.lisp [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+            (cl:when (cl:zerop (cl:mod bi 100))
+              (cl:format cl:t ";; ~D~%" bi))
+            Unit)
+      (for ai in (iter:up-to steps-num)
+        (let a = (+ real-offset (* width (/ ai steps-num)))
+          )
```
Comment (stylewarning) [inline benchmarks/benchmarks-game/mandelbrot.lisp:85]: fix parens

File: benchmarks/benchmarks-game/too-simple.lisp [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+   #:too-simple-main2))
+
+(in-package #:coalton/benchmarks/benchmarks-game/too-simple)
+
+(cl:declaim (cl:optimize cl:speed (cl:safety 0) (cl:debug 0)))
+(coalton-toplevel
+
+  (define (too-simple-main n)
+    (time (fn ()
+            (let sum = (cell:new 0.0d0))
```
Comment (stylewarning) [inline benchmarks/benchmarks-game/too-simple.lisp:20]: lift the fn out into its own function

File: benchmarks/benchmarks-game/too-simple.lisp [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+              (cell:update! negate flip)
+              (cell:update! (fn (x) (+ x
+                                        (/ (cell:read flip)
+                                           (1- (* 2 i)))))
+                            sum))
+            (traceobject "Sum" (* (cell:read sum) 4)))))
+
+  (define (too-simple-main2 n)
+    (time (fn ()
+            (let ((declare run (Double-Float -> Double-Float -> Double-Float -> Double-Float))
```
Comment (stylewarning) [inline benchmarks/benchmarks-game/too-simple.lisp:32]: same

File: benchmarks/big-float.lisp [hunk 7]
```diff
@@ -1,17 +1,68 @@
...
+        (let x = (into x))
+        (sqrt x))))
+
+  (define (big-mult-constants n x)
+    (with-precision n
+      (fn ()
+        (let x = (into x))
+        (* x (* pi ee))))))
+
+;;(cl:in-package #:coalton-benchmarks)
```
Comment (stylewarning) [inline benchmarks/big-float.lisp:55]: dead code

File: benchmarks/big-float.lisp [hunk 8]
```diff
@@ -1,17 +1,68 @@
...
+
+#+ig
```
Comment (stylewarning) [inline benchmarks/big-float.lisp:57]: figure out what to do with these

File: benchmarks/fibonacci.lisp [hunk 9]
```diff
@@ -2,31 +2,89 @@
...
+  (define (fib-monomorphized-optional x)
+    (fib-generic (Some x))))
+
+;;;
+;;; Deprecated Trivial-Benchmarks
+;;;
+
+;;(cl:in-package #:coalton-benchmarks)
+
+#+ig (define-benchmark recursive-fib ()
```
Comment (stylewarning) [inline benchmarks/fibonacci.lisp:65]: figure out what to do here.

**Review by stylewarning — CHANGES_REQUESTED** [review]
this PR is still a draft and needs a lot of preparation

**Comment by Izaakwltn** [comment]
Whoops, I thought I marked it as a draft

**Comment by Izaakwltn** [comment]
Closing this as it's superseded by #1286 

The additional benchmarks defined in this PR will be added to a new PR after #1286 merges.

**Pull request closed without merging**